### ISSUE TITLE
NetworkActivityIndicator running on a background thread

### ISSUE THREAD

**Issue by arctouch-carlosvinicius** [mixpanel/mixpanel-swift#134]
Xcode: 9.0
version: 2.1.2
device: iPhone 7+, iOS 11

Xcode 9 is pointing out that NetworkActivityIndicator is running on a background thread. 
I've found a similar issue here: https://github.com/mixpanel/mixpanel-iphone/issues/677

[reports] Main Thread Checker: UI API called on a background thread: -[UIApplication setNetworkActivityIndicatorVisible:]
PID: 612, TID: 106905, Thread name: (none), Queue name: com.mixpanel.e0e109a641b99848100a94b8bfc6e61b, QoS: 0
Backtrace:
4   Mixpanel                            0x0000000105cfb30c _T08Mixpanel0A8InstanceC30updateNetworkActivityIndicatorySbF + 188
5   Mixpanel                            0x0000000105cfbe54 _T08Mixpanel0A8InstanceCAA13FlushDelegateA2aDP30updateNetworkActivityIndicatorySbFTW + 68
6   Mixpanel                            0x0000000105cc2b38 _T08Mixpanel5FlushC19flushQueueInBatchesySays10DictionaryVySSypGGz_AA0B4TypeO4typetF + 1160
7   Mixpanel                            0x0000000105cc1d48 _T08Mixpanel5FlushC10flushQueueyAA0B4TypeO4type_Says10DictionaryVySSypGGz5queuetF + 176
8   Mixpanel                            0x0000000105cc1c3c _T08Mixpanel5FlushC16flushEventsQueueySays10DictionaryVySSypGGzF + 68
9   Mixpanel                            0x0000000105cfe020 _T08Mixpanel0A8InstanceC5flushyyycSg10completion_tFyycfU_ + 548
10  Mixpanel                            0x0000000105c9acb4 _T0Ix_IyB_TR + 48
11  libdispatch.dylib                   0x000000010733149c _dispatch_call_block_and_release + 24
12  libdispatch.dylib                   0x000000010733145c _dispatch_client_callout + 16
13  libdispatch.dylib                   0x0000000107340110 _dispatch_queue_serial_drain + 692
14  libdispatch.dylib                   0x00000001073349a4 _dispatch_queue_invoke + 332
15  libdispatch.dylib                   0x0000000107341104 _dispatch_root_queue_drain_deferred_wlh + 424
16  libdispatch.dylib                   0x0000000107348100 _dispatch_workloop_worker_thread + 652
17  libsystem_pthread.dylib             0x0000000186dbafe0 _pthread_wqthread + 932
18  libsystem_pthread.dylib             0x0000000186dbac30 start_wqthread + 4

**Comment by MihaelIsaev** [comment]
same for me

```javascript
Main Thread Checker: UI API called on a background thread: -[UIApplication setNetworkActivityIndicatorVisible:]
PID: 11081, TID: 21785716, Thread name: (none), Queue name: com.mixpanel.c5d6567504c25489937e087c3b535bc3.0x7fb00a10a040, QoS: 0
Backtrace:
4   Mixpanel                            0x00000001025a6253 -[MPNetwork updateNetworkActivityIndicator:] + 82
5   Mixpanel                            0x00000001025a4ceb -[MPNetwork flushQueue:endpoint:] + 449
6   Mixpanel                            0x000000010258b169 __32-[Mixpanel flushWithCompletion:]_block_invoke + 199
7   libdispatch.dylib                   0x000000010a0e13f7 _dispatch_call_block_and_release + 12
8   libdispatch.dylib                   0x000000010a0e243c _dispatch_client_callout + 8
9   libdispatch.dylib                   0x000000010a0ea95b _dispatch_queue_serial_drain + 1162
10  libdispatch.dylib                   0x000000010a0eb2df _dispatch_queue_invoke + 336
11  libdispatch.dylib                   0x000000010a0e707d _dispatch_queue_override_invoke + 733
12  libdispatch.dylib                   0x000000010a0ee1f9 _dispatch_root_queue_drain + 772
13  libdispatch.dylib                   0x000000010a0ede97 _dispatch_worker_thread3 + 132
14  libsystem_pthread.dylib             0x000000010a6765a2 _pthread_wqthread + 1299
15  libsystem_pthread.dylib             0x000000010a67607d start_wqthread + 13
```

**Comment by th3m477** [comment]
Same, probably needs fixing in main branch as well as swift4 version?

**Comment by yarneo** [comment]
this has been fixed in the latest release ( v2.2.2 ) , thanks!

**Comment by andrewnagata** [comment]
I am seeing this error on version 3.1.2 from fresh cocoapods install
